plugins/link/SharedMemory.cpp

```diff
@@ -146,11 +150,47 @@ void *SharedMemory::mapMemory(const char *name, std::size_t size) {
 	m_name.assign(name);
 #endif
 
-	m_size = size;
-
 	if (created) {
-		std::memset(m_data, 0, m_size);
+		fillWithZero();
+	}
+
+	return true;
+}
+
+bool SharedMemory::isMemoryMapped() {
+	return m_data != nullptr;
+}
+
+void SharedMemory::read(LinkedMem *dest) {
+	std::atomic< Chunk > *data = (std::atomic< Chunk > *) m_data;
```

Comment:
```suggestion
	std::atomic< Chunk > *data = reinterpret_cast< std::atomic< Chunk > *>(m_data);
```
Is this a safe thing to do? In general I would assume that we at least have to ensure the size of `m_data` (the array/buffer) is a multiple of the size of `std::atomic<Chunk>` or rather - since we have already enforced that it is a multiple of `Chunk` - that `sizeof(std:.atomic<Chunk> == sizeof(Chunk)`.

Reply:
I would use `static_cast()` instead.

Reply:
Static casts can't be used on pointers

Reply:
Are you sure? I used static_cast, and it seems to work. cppreference.com says static_cast can be used to convert a void pointer to a pointer of any object type.

Reply:
Correct.

Reply:
Oh right - it's a void pointer. Yeah, then it works and `static_cast` should indeed be preferred.